Sum of digits of a number.py

```diff
@@ -1,13 +1,34 @@
-q=0   # Initially we assigned 0 to "q", to use this variable for the summation purpose below.
-    # The "q" value should be declared before using it(mandatory). And this value can be changed later.
+# Python code to calculate the sum of digits of a number, by taking number input from user.
 
-n=int(input("Enter Number: "))   # asking user for input
-while n>0:         # Until "n" is greater than 0, execute the loop. This means that until all the digits of "n" got extracted.
+import sys
 
-   r=n%10           # Here, we are extracting each digit from "n" starting from one's place to ten's and hundred's... so on.
+def get_integer():
+    for i in range(3,0,-1):                       # executes the loop 3 times. Giving 3 chances to the user.
+        num = input("enter a number:")
+        if num.isnumeric():                       # checks if entered input is an integer string or not.
+            num = int(num)                        # converting integer string to integer. And returns it to where function is called.
+            return num
+        else:
+            print("enter integer only")                    
+            if (i-1)<=1:                                   # if user entered wrong input, it displays a message and chances left.
+                print(f'{i-1} chance is left')             
+            else:
+                print(f'{i-1} chances are left')
+         
+        continue   
+                
+def addition(num):
+    Sum=0
+    if type(num) is type(None):               # Checks if number type is none or not. If type is none program exits.
+        print("Try again!")
+        sys.exit()
+    while num > 0:                            # Addition- adding the digits in the number.
+        digit = int(num % 10)
+        Sum += digit
```

Comment:
Also use doc-string on the main function.
To explain it when.
print(function.\_\_doc\_\_  ) is used.

Reply:
why this? @NitkarshChourasia

Reply:
> Also use doc-string on the main function. To explain it when. print(function.__doc__ ) is used.

why this?